context.go

```diff
@@ -125,15 +128,27 @@ type WSHandshake func(*websocket.Config, *http.Request) error
 func (context *Context) WebSocketHandshake() WSHandshake {
 	return func(config *websocket.Config, rq *http.Request) error {
 		protocolHeader := rq.Header.Get("Sec-WebSocket-Protocol")
-		if !includesProtocol(protocolHeader, context.WebSocketSubProtocol) {
-			context.log("Error: Client doesn't support WS protocol %s, only %s", context.WebSocketSubProtocol, protocolHeader)
-			return &websocket.ProtocolError{"I only speak " + context.WebSocketSubProtocol + " protocol"}
+
+		protocol, found := includesProtocol(protocolHeader, context.SupportedSubProtocols)
+
+		if !found {
+			stringSeperatedProtocols := strings.Join(context.SupportedSubProtocols, ",")
+			context.log("Error: Client doesn't support any of WS protocols: %s only %s", stringSeperatedProtocols, protocolHeader)
+			return &websocket.ProtocolError{
+				ErrorString: "I only speak " + stringSeperatedProtocols + " protocols",
+			}
 		}
-		config.Protocol = []string{context.WebSocketSubProtocol}
+
+		config.Protocol = []string{protocol}
+		context.activeSubProtocol = protocol
 		return nil
 	}
 }
 
+func (context *Context) ActiveProtocol() string {
```

Comment:
Comment on public symbols please. Need something visible for GoDoc.

```suggestion
// ActiveProtocol returns the currently used WebSocket subprotocol for the Context, set after a successful handshake.
func (context *Context) ActiveProtocol() string {
```